### DISCUSSION TITLE
nx g @nx/vue:setup-unocss

### DISCUSSION THREAD

**Discussion by gnuion** [nrwl/nx#23114] [Feature Requests]
setup unocss for vue3 application

**Comment by jaysoo** [comment]
We generally want to reduce the amount of styling options we support out of the box. Solutions in the styling space change and evolve over time, and we don't want to commit too much resource to keep them up to date.

Unocss doesn't require that much setup to warrant its own generator. From what I can tell it is:

1. Install the plugin npm package
2. Update `vite.config.ts` to include the plugin

If I'm missing something please let me know what exactly this generator would do in addition to the above two things.

**Reply by gnuion** [reply]
correct, unocss does not require much setup. tailwind is also easy. i would agree that a generator is most useful for complex setups that are difficult to setup from scratch each time

**Reply by jaysoo** [reply]
If there is enough interest we'll look into it. It doesn't have that much usage yet (according to https://www.npmjs.com/package/unocss).
